components/homme/src/share/gllfvremap_mod.F90

```diff
@@ -0,0 +1,2253 @@
+                  gfr%qmax(k,qi,ie), dp(:,:,k), elem(ie)%derived%FQ(:,:,k,qi))
+          end do
+          if (gfr%check) then
+             call check_f2g_mixing_ratio(gfr, hybrid, ie, qi, elem, gfr%qmin(:,qi,ie), &
+                  gfr%qmax(:,qi,ie), dp, wr1, elem(ie)%derived%FQ(:,:,:,qi))
+          end if
+          if (.not. q_adjustment) then
+             ! Convert to a tendency.
+             elem(ie)%derived%FQ(:,:,:,qi) = &
+                  dp*(elem(ie)%derived%FQ(:,:,:,qi) - elem(ie)%state%Q(:,:,:,qi))/dt
+          end if
+       end do
+    end do
+  end subroutine gfr_fv_phys_to_dyn_hybrid
+
+  subroutine gfr_dyn_to_fv_phys_topo(hybrid, elem, nets, nete, phis)
+    ! If needed, remap topography data defined on the GLL grid to the
+    ! FV grid. The intended EAM configuration is to use topography
+    ! data on the FV grid, so this routine is unlikely to be used.
+
+    type (hybrid_t), intent(in) :: hybrid
+    type (element_t), intent(in) :: elem(:)
+    integer, intent(in) :: nets, nete
+    real(kind=real_kind), intent(out) :: phis(:,:)
+
+    integer :: ie
+
+    do ie = nets,nete
+       call gfr_dyn_to_fv_phys_topo_elem(elem, ie, phis(:,ie))
+    end do
+  end subroutine gfr_dyn_to_fv_phys_topo
+
+  subroutine gfr_fv_phys_to_dyn_topo_hybrid(hybrid, elem, nets, nete, phis)
+    ! Remap FV topography data to the GLL grid. Prevent new
+    ! extrema. Conserve the integral of height.
+
+    use kinds, only: iulog
+    use edge_mod, only: edgeVpack_nlyr, edgeVunpack_nlyr, edge_g
+    use bndry_mod, only: bndry_exchangeV
+
+    type (hybrid_t), intent(in) :: hybrid
+    type (element_t), intent(inout) :: elem(:)
+    integer, intent(in) :: nets, nete
+    real(kind=real_kind), intent(in) :: phis(:,:)
+
+    real(kind=real_kind) :: wr(np,np,2), ones(np,np,1)
+    integer :: ie, nf, ncol
+
+    ones = one
+    nf = gfr%nphys
+    ncol = nf*nf
+    ! For now, map GLL topo back to FV, as ne30pg2 runs otherwise show grid
+    ! imprint in the PHIS diagnostic. We may revise this choice later.
+    !gfr%have_fv_topo_file_phis = .true.
+
+    do ie = nets,nete
+       gfr%phis(:,ie) = phis(:ncol,ie)
+       wr(:nf,:nf,1) = reshape(phis(:ncol,ie), (/nf,nf/))
+       gfr%qmin(:,:,ie) = minval(wr(:nf,:nf,1))
+       gfr%qmax(:,:,ie) = maxval(wr(:nf,:nf,1))
+       if (nf > 1) then
+          call gfr_f2g_scalar(ie, elem(ie)%metdet, wr(:,:,1:1), wr(:,:,2:2))
+          elem(ie)%state%phis = wr(:,:,2)
+       else
+          elem(ie)%state%phis = wr(1,1,1)
+       end if
+    end do
+
+    if (hybrid%par%dynproc) then
+       call gfr_f2g_mixing_ratios_he(hybrid, nets, nete, gfr%qmin(:,:,nets:nete), &
+            gfr%qmax(:,:,nets:nete))
+    end if
+
+    if (nf > 1 .or. .not. gfr%boost_pg1) then
+       do ie = nets,nete
+          if (gfr%check) wr(:,:,1) = elem(ie)%state%phis
+          call limiter_clip_and_sum(np, elem(ie)%spheremp, gfr%qmin(1,1,ie), &
```

Comment:
you limit topo?

Reply:
Yes. This is up for discussion, along with everything else we're thinking about re: topo. But for now I feel it's safest not to introduce new topo extrema. So the topo remap problem is: remap such that the average height is conserved and so that no new extrema are created.